### ISSUE TITLE
Bad codegen with struct alignment and `offsetof`

### ISSUE THREAD

**Issue by JohanEngelen** [ldc-developers/ldc#2346]
~~Although the exact alignment requirements are unclear~~
The exact alignment requirements are clear since 2.075 (see forum post: http://forum.dlang.org/post/vzvxkabifknrzffycjao@forum.dlang.org), the following code should not crash but does.

```d
struct UInt {
align(1):
    uint a;
}

struct Bug {
    ubyte one;
    UInt two;
}

align(1)
struct Align1UInt {
align(1):
    uint a;
}

struct BugAlign1 {
    ubyte one;
    Align1UInt two;
}

version(CRASH_DURING_COMPILATION) {
    struct A {
        Bug a;
        BugAlign1 b;
    }
}

void foobug(Bug* b) {
    b.two.a = 42;

    void* ptr = cast(void*)b;
    ptr += Bug.two.offsetof;
    assert( ptr == &b.two );
    assert( (cast(UInt*)ptr).a == 42 );
}

void fooalign1bug(BugAlign1 *b) {
    b.two.a = 42;

    void* ptr = cast(void*)b;
    ptr += BugAlign1.two.offsetof;
    assert( ptr == cast(void*)&b.two );
    assert( (cast(UInt*)ptr).a == 42 );
}

void main() {
    Bug b;
    foobug(&b);

    BugAlign1 b2;
    fooalign1bug(&b2);
}
```

DMD runs the code fine.

**Comment by JohanEngelen** [comment]
We create the LLVM type wrong for `Bug` and `BugAlign1` (they should be the same btw).

We almost output things correctly, I seems.
Slightly more complex case:
```d
struct UInt {
align(2):
    uint a;
}

struct Bug {
    ubyte one;
    UInt two;
}
```
We output 
```llvm
%agrsize.UInt = type { i32 }
%agrsize.Bug = type { i8, [1 x i8], %agrsize.UInt }
```
(note that we already add the padding)
but I think it should be 
```llvm
%agrsize.UInt = type { i32 }
%agrsize.Bug = type <{ i8, [1 x i8], %agrsize.UInt }>
```
(angled brackets to indicate packed type, and that we ourselves take care of the padding)

**Comment by kinke** [comment]
I have a fix I think, PR is coming up shortly.

**Comment by JohanEngelen** [comment]
Fixed by #2347
